near-contract-standards/src/non_fungible_token/event.rs

```diff
@@ -0,0 +1,174 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "standard")]
+#[serde(rename_all = "snake_case")]
+pub enum NearEvent {
+    Nep171(Nep171Event),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Nep171Event {
+    pub version: String,
+    #[serde(flatten)]
+    pub event_kind: Nep171EventKind,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "event", content = "data")]
+#[serde(rename_all = "snake_case")]
+#[allow(clippy::enum_variant_names)]
+pub enum Nep171EventKind {
+    NftMint(Vec<NftMintData>),
+    NftTransfer(Vec<NftTransferData>),
+    NftBurn(Vec<NftBurnData>),
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftMintData {
+    pub owner_id: String,
+    pub token_ids: Vec<String>,
+    pub memo: Option<String>,
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftTransferData {
+    pub authorized_id: Option<String>,
+    pub old_owner_id: String,
+    pub new_owner_id: String,
+    pub token_ids: Vec<String>,
+    pub memo: Option<String>,
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftBurnData {
+    pub authorized_id: Option<String>,
+    pub owner_id: String,
+    pub token_ids: Vec<String>,
+    pub memo: Option<String>,
+}
```

Comment:
We may consider using [`Cow<'a, str>`](https://deterministic.space/secret-life-of-cows.html) instead of Strings to avoid unnecessary clones when users would need to generate the events without giving the ownership over the strings. Yet, that might make the API harder to follow.

Reply:
Yeah again this is a copy/paste and I agree that we should try to minimize the cost of these logs.